src/prognostic_equations/implicit/implicit_solver.jl

```diff
@@ -677,19 +697,28 @@ function update_implicit_equation_jacobian!(A, Y, p, dtγ)
             @. ∂ᶜuₕ_err_∂ᶜuₕ =
                 dtγ * DiagonalMatrixRow(1 / ᶜρ) ⋅ ᶜdiffusion_u_matrix - (I,)
         end
+    end
 
+    if p.atmos.precip_model isa Microphysics1Moment
         ᶠlg = Fields.local_geometry_field(Y.f)
         precip_info =
             ((@name(c.ρq_rai), @name(ᶜwᵣ)), (@name(c.ρq_sno), @name(ᶜwₛ)))
         MatrixFields.unrolled_foreach(precip_info) do (ρqₚ_name, wₚ_name)
-            MatrixFields.has_field(Y, ρqₚ_name) || return
             ∂ᶜρqₚ_err_∂ᶜρqₚ = matrix[ρqₚ_name, ρqₚ_name]
+            ∂ᶜρqₚ_err_∂ᶜρ = matrix[ρqₚ_name, @name(c.ρ)]
             ᶜwₚ = MatrixFields.get_field(p, wₚ_name)
-            ᶠtmp = p.ᶠtemp_CT3
-            @. ᶠtmp = CT3(unit_basis_vector_data(CT3, ᶠlg)) * ᶠwinterp(ᶜJ, ᶜρ)
-            @. ∂ᶜρqₚ_err_∂ᶜρqₚ +=
-                dtγ * -(ᶜprecipdivᵥ_matrix()) ⋅ DiagonalMatrixRow(ᶠtmp) ⋅
-                ᶠright_bias_matrix() ⋅ DiagonalMatrixRow(-(ᶜwₚ) / ᶜρ)
+            if use_derivative(diffusion_flag)
+                @. ∂ᶜρqₚ_err_∂ᶜρqₚ +=
+                    dtγ * -(ᶜprecipdivᵥ_matrix()) ⋅
+                    DiagonalMatrixRow(Geometry.WVector(1) * ᶠwinterp(ᶜJ, ᶜρ)) ⋅
+                    ᶠright_bias_matrix() ⋅ DiagonalMatrixRow(-(ᶜwₚ) / ᶜρ)
+            else
+                @. ∂ᶜρqₚ_err_∂ᶜρqₚ =
+                    dtγ * -(ᶜprecipdivᵥ_matrix()) ⋅
+                    DiagonalMatrixRow(Geometry.WVector(1) * ᶠwinterp(ᶜJ, ᶜρ)) ⋅
+                    ᶠright_bias_matrix() ⋅ DiagonalMatrixRow(-(ᶜwₚ) / ᶜρ) - (I,)
+                ∂ᶜρqₚ_err_∂ᶜρ .= (zero(eltype(∂ᶜρqₚ_err_∂ᶜρ)),)
```

Comment:
Since we're not planning to add these blocks later, we can just get rid of this. Hopefully it doesn't change anything.